testing/go/types_test.go

```diff
@@ -1283,6 +1287,257 @@ var typesTests = []ScriptTest{
+				Expected: []sql.Row{
+					{123, 123, 0, 400},
+				},
+			},
+			{
+				Query: "SELECT ('-1'::char(3))::xid, ('-1'::varchar)::xid, ('-1'::text)::xid, ('-1'::name)::xid;",
+				Expected: []sql.Row{
+					{4294967295, 4294967295, 4294967295, 4294967295},
+				},
+			},
+			{
+				Query: "SELECT ('-2147483648'::char(11))::xid, ('-2147483648'::varchar)::xid, ('-2147483648'::text)::xid, ('-2147483648'::name)::xid;",
+				Expected: []sql.Row{
+					{2147483648, 2147483648, 2147483648, 2147483648},
+				},
+			},
+			{
+				Query:       "SELECT (10::int2)::xid;",
+				ExpectedErr: true,
+			},
+			{
+				Query:       "SELECT (10::boolean)::xid;",
+				ExpectedErr: true,
+			},
+			{
+				Query:       "SELECT (10::int4)::xid;",
+				ExpectedErr: true,
+			},
+			{
+				Query:       "SELECT (10::int8)::xid;",
+				ExpectedErr: true,
+			},
+			{
+				Query:       "SELECT (1.1::float4)::xid;",
+				ExpectedErr: true,
+			},
+			{
+				Query:       "SELECT (1.1::float8)::xid;",
+				ExpectedErr: true,
+			},
+			{
+				Query:       "SELECT (1.1::decimal)::xid;",
+				ExpectedErr: true,
+			},
+			{
+				Query: "SELECT ('4294967295'::text)::xid, ('4294967297'::text)::xid;",
+				Expected: []sql.Row{
+					{4294967295, 1},
+				},
+			},
+			{
+				Query: "SELECT ('-4294967295'::text)::xid, ('-4294967297'::text)::xid;",
+				Expected: []sql.Row{
+					{1, 4294967295},
+				},
+			},
+			{
+				Query: "SELECT ('4294967295'::varchar)::xid, ('4294967296232'::varchar)::xid;",
+				Expected: []sql.Row{
+					{4294967295, 232},
+				},
+			},
+			{
+				Query: "SELECT ('-4294967295'::varchar)::xid, ('-4294967296232'::varchar)::xid;",
+				Expected: []sql.Row{
+					{1, 4294967064},
+				},
+			},
+			{
+				Query: "SELECT ('4294967295'::char(11))::xid, ('4294967296'::char(11))::xid;",
+				Expected: []sql.Row{
+					{4294967295, 0},
+				},
+			},
+			{
+				Query: "SELECT ('4294967295'::name)::xid, ('4294967296'::name)::xid;",
+				Expected: []sql.Row{
+					{4294967295, 0},
+				},
+			},
+			{
+				Query: "SELECT ('abc'::text)::xid, ('abc'::char(3))::xid, ('abc'::varchar)::xid, ('abc'::name)::xid;",
+				Expected: []sql.Row{
+					{0, 0, 0, 0},
+				},
+			},
+		},
+	},
+	{
+		Name: "Xid array type",
+		SetUpScript: []string{
+			"CREATE TABLE t_xid (id INTEGER primary key, v1 XID[], v2 CHARACTER(100), v3 BOOLEAN);",
+			// TODO: Insert to XID[] column not working (invalid type: xid[])
+			// "INSERT INTO t_xid VALUES (2, '{123, 456, 789, 101}', '1234567890', true);",
+		},
+		Assertions: []ScriptTestAssertion{
+			// {
+			// 	Query: `SELECT v1::varchar(1)[] FROM t_xid;`,
+			// 	Expected: []sql.Row{
+			// 		{"{1,4,7,1}"},
+			// 	},
+			// },
+			// {
+			// 	Query: `SELECT v2::xid, v3::xid FROM t_xid;`,
+			// 	Expected: []sql.Row{
+			// 		{1234567890, 1},
+			// 	},
+			// },
+			// {
+			// 	Query:       `INSERT INTO t_xid VALUES (2, ARRAY[123, 456, 789, 101], '1234567890', true);`,
+			// 	ExpectedErr: true,
+			// },
+		},
+	},
```

Comment:
We can uncomment this and leave it as a skipped test. This is hitting the `sql.Type.Convert` code in `server/types.array.go` and throwing an error as it expects values of type `[]any` but it's receiving a string. This will be taken care of with the `Convert` removal.